### ISSUE TITLE
jinja[invalid] false positive on loop variable with ansible.utils.ipv4 filter

### ISSUE THREAD

**Issue by Urth** [ansible/ansible-lint#2491]
##### Summary

Using the loop variable `{{ ip_list | ansible.utils.ipv4 }}` triggers the `jinja[invalid]` linting rule.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
$ ansible --version
ansible [core 2.13.3]
  config file = /home/harm/dev/closso-ansible/dostno.cfg
  configured module search path = ['/home/harm/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/harm/.virtualenvs/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/harm/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/harm/.virtualenvs/ansible/bin/ansible
  python version = 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
  jinja version = 3.1.2
  libyaml = True

$ ansible-lint --version
ansible-lint 6.6.1 using ansible 2.13.3
&
ansible-lint 6.6.2.dev5 using ansible 2.13.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

```console
$ lsb_release -a 
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```

##### STEPS TO REPRODUCE

```yml
- name: Test jinja invalid
  hosts: all
  vars:
    ip_list:
      - 8.8.8.8
      - 8.8.4.4
      - 2001:4860:4860::8888
      - 2001:4860:4860::8844
  tasks:
    - name: Loop filter is seen as invalid
      debug:
        var: item
      loop: "{{ ip_list | ansible.utils.ipv4 }}"
```

##### Desired Behavior

The loop variable should be allowed.

##### Actual Behavior

```console
$ ansible-lint -vvvv jinja_invalid.yml
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/home/harm/.cache/ansible-compat/fa3aad', colored=False, configured=True, cwd=PosixPath('/home/harm/dev/closso-ansible'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format='rich', lintables=['jinja_invalid.yml'], listrules=False, listtags=False, write_list=[], parseable=True, quiet=True, rulesdirs=['/home/harm/.virtualenvs/ansible/lib/python3.10/site-packages/ansiblelint/rules'], skip_list=['empty-string-compare', 'fqcn-builtins', 'name[casing]', 'schema', 'yaml[line-length]'], tags=[], verbosity=3, warn_list=['experimental', 'jinja[spacing]', 'name[casing]', 'name[play]', 'role-name'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.{yaml,yml}'}, {'requirements': '**/meta/requirements.{yaml,yml}'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=False, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, rules={}, profile=None, progressive=False, rulesdir=[], use_default_rules=False, config_file='/home/harm/dev/closso-ansible/.ansible-lint', version=False, cache_dir_lock=<filelock._unix.UnixFileLock object at 0x7f426898cb50>)
DEBUG    /home/harm/dev/closso-ansible
INFO     Running ansible-galaxy collection install -vr requirements.yml -p /home/harm/.cache/ansible-compat/fa3aad/collections
INFO     Set ANSIBLE_LIBRARY=/home/harm/.cache/ansible-compat/fa3aad/modules:/home/harm/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/harm/.cache/ansible-compat/fa3aad/collections:/home/harm/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/harm/.cache/ansible-compat/fa3aad/roles:roles:/home/harm/dev/closso-ansible/roles
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
INFO     Set ANSIBLE_LIBRARY=/home/harm/.cache/ansible-compat/fa3aad/modules:/home/harm/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/harm/.cache/ansible-compat/fa3aad/collections:/home/harm/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/harm/.cache/ansible-compat/fa3aad/roles:roles:/home/harm/dev/closso-ansible/roles
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: jinja_invalid.yml
INFO     Executing syntax check on jinja_invalid.yml (1.35s)
DEBUG    Examining jinja_invalid.yml of type playbook
WARNING  Listing 1 violation(s) that are fatal
jinja_invalid.yml:10: jinja (jinja[invalid])
DEBUG    Attempting to release lock 139923199413072 on /home/harm/.cache/ansible-compat/fa3aad/.lock
DEBUG    Lock 139923199413072 released on /home/harm/.cache/ansible-compat/fa3aad/.lock
```

```console
$ ansible-playbook jinja_invalid.yml -C -i localhost,
PLAY [Test jinja invalid] ***********************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Loop filter is seen as invalid] ***********************************************************************************************************************************************************************************************************
ok: [localhost] => (item=8.8.8.8) => {
    "ansible_loop_var": "item",
    "item": "8.8.8.8"
}
ok: [localhost] => (item=8.8.4.4) => {
    "ansible_loop_var": "item",
    "item": "8.8.4.4"
}

PLAY RECAP **************************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Jakob3xD** [comment]
Have a similar issue.
ansible-lint:
```
jinja: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipaddr filter <value> (jinja[invalid])
```
play/task:
```
value: "{{ k8s_api_floating_ipv6 ansible.utils.ipaddr('address') }}"
```
workaround:
```
value: "{{ k8s_api_floating_ipv6 | default('') | ansible.utils.ipaddr('address') }}"
```

Defining the var under extra_vars in the ansible-lint config does not help.

**Comment by ssbarnea** [comment]
I am using the code from main branch I am unable to get an error, only a runtime warning:

```
[WARNING]: an unexpected error occurred during Jinja2 environment setup: unable to locate collection ansible.utils
```
That does not produce any rule violation, but probably it should because `ansible.utils` collection is indeed not installed and it should be declared as needed via a `requirements.yml` file. Still, is not really a bug but more of a missing feature.

I know that at some point it was an error but I decided to make it runtime only because it was confusing too many people.

Closing as fixed in v6.7.0 but ping me if you can still reproduce it. Also mention if adding the missing collection to to requirements file does prevent it from happening.

**Comment by kpfleming** [comment]
I just started seeing this today after upgrading to 6.7.0, and adding `ansible.utils` under `collections` in `requirements.yml` does not fix it. Here's an example error message:

```
jinja: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipwrap filter <value> (jinja[invalid])
roles/stork/tasks/server.yml:53 Task/Handler: set_fact stork_url={{ 'http://' ~ host_ipv6|ansible.utils.ipwrap ~ ':8989' }}
```

**Comment by kpfleming** [comment]
Looks the same as #2468.
